crates/katana/primitives/src/utils/transaction.rs

```diff
@@ -382,4 +382,270 @@ mod tests {
+            felt!("0x1")
+        ];
+        let class_hash = felt!("0x036078334509b514626504edc9fb252328d1a240e4e948bef8d0c08dff45927f");
+        let contract_address_salt = felt!("0x74a02936feda8279d6df2c6ca0991281674fa028fed1990ad9ad460509fa411");
+        let tip = 0;
+        let l1_gas_bounds = ResourceBounds {
+            max_amount: 0x29,
+            max_price_per_unit: 0x16b812d3fa41,
+        };
+        let l2_gas_bounds = ResourceBounds {
+            max_amount: 0x0,
+            max_price_per_unit: 0x0,
+        };
+        let paymaster_data = vec![];
+        let chain_id = ChainId::MAINNET.id();
+        let nonce = felt!("0x0");
+        let nonce_da_mode = &DataAvailabilityMode::L1;
+        let fee_da_mode = &DataAvailabilityMode::L1;
+    
+        let actual_hash = compute_deploy_account_v3_tx_hash(
+            contract_address,
+            &constructor_calldata,
+            class_hash,
+            contract_address_salt,
+            tip,
+            &l1_gas_bounds,
+            &l2_gas_bounds,
+            &paymaster_data,
+            chain_id,
+            nonce,
+            nonce_da_mode,
+            fee_da_mode,
+            false,
+        );
+    
+        assert_eq!(actual_hash, expected_hash);
+    }
+
+    #[test]
+    fn test_compute_declare_v1_tx_hash() {
+        // Starknet mainnet tx hash: https://voyager.online/tx/0x1f01dd957c47a81ce2be2426770693ffb7a155e54f9c556c40b943ce88d1859
+        let expected_hash = felt!("0x1f01dd957c47a81ce2be2426770693ffb7a155e54f9c556c40b943ce88d1859");
+
+        let sender_address = felt!("0x4d2c7d94a05cd95e08f1c135c53aa798f26ac383198d77bd37822e646cbab44");
+        let class_hash = felt!("0xd0879f156c3e060638d5fb8ea1604cada1a29017988b3ee4f5f8b653279f60");
+        let max_fee = felt!("0x1cfe57d53f9f");
+        let chain_id = ChainId::MAINNET.id();
+        let nonce = felt!("0xb");
+
+        let actual_hash = compute_declare_v1_tx_hash(
+            sender_address,
+            class_hash,
+            max_fee.to_u128().unwrap(),
+            chain_id,
+            nonce,
+            false,
+        );
+    
+        assert_eq!(actual_hash, expected_hash);
+    }
+
+    #[test]
+    fn test_compute_declare_v2_tx_hash() {
+        // Starknet mainnet tx hash: https://voyager.online/tx/0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9
+        let expected_hash = felt!("0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9");
+
+        let sender_address = felt!("0x020c398d72af5efa4b63f5e3d5ad21e981d6af5f5929cfd2ab0d759ff935be53");
+        let class_hash = felt!("0x0311b6f080fd3385e7154ca3a8568eb7d6aebcb7ff627c1f5e7d2cc99aeb7741");
+        let max_fee = felt!("0x108ae97efa9f8");
+        let chain_id = ChainId::MAINNET.id();
+        let nonce = felt!("0xb");
+        let compiled_class_hash = felt!("0x311b6f080fd3385e7154ca3a8568eb7d6aebcb7ff627c1f5e7d2cc99aeb7741");
```

Comment:
the compiled class hash here should be `0x29b2702c06c1e3f3fe79e5b5e89071e9c4a8e82955a633a3879e3fae1dd7c3c`.

```sh
curl -X POST https://starknet-mainnet.public.blastapi.io -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"starknet_getTransactionByHash","params":[ "0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9" ],"id":1}' | jq
{
  "jsonrpc": "2.0",
  "result": {
    "transaction_hash": "0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9",
    "type": "DECLARE",
    "version": "0x2",
    "nonce": "0xb",
    "max_fee": "0x108ae97efa9f8",
    "class_hash": "0x311b6f080fd3385e7154ca3a8568eb7d6aebcb7ff627c1f5e7d2cc99aeb7741",
    "sender_address": "0x20c398d72af5efa4b63f5e3d5ad21e981d6af5f5929cfd2ab0d759ff935be53",
    "signature": [
      "0x40a0ab8b0d0a9cd8e97a67178bb449bd5f404b8335db96ea9b9567c348a96",
      "0x57595113723c85653bdcfa46d9928a1dc5c65c44f7430e567939a67e60b8dd7"
    ],
    "compiled_class_hash": "0x29b2702c06c1e3f3fe79e5b5e89071e9c4a8e82955a633a3879e3fae1dd7c3c"
  },
  "id": 1
}
```